chainerx_cc/chainerx/float16_test.cc

```diff
@@ -0,0 +1,168 @@
+#include "chainerx/float16.h"
+
+#include <cmath>
+#include <limits>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+namespace chainerx {
+namespace {
+
+void CheckNear(double expected, double actual, double tol) {
+    if (std::isnan(expected)) {
+        // NaN
+        EXPECT_TRUE(std::isnan(actual));
+    } else if (std::isinf(expected)) {
+        // Signed inf
+        EXPECT_EQ(expected, actual);
+    } else {
+        // Absolute error or relative error should be less or equal to tol.
+        tol = std::max(tol, tol * std::abs(expected));
+        EXPECT_NEAR(expected, actual, tol);
+    }
+}
+
+void CheckFloat16(double x, double tol) {
```

Comment:
This test checks the combinations of ctors and conversion operators. Isn't it possible to separate them? Also, please name the function accordingly based on the checks, and/or add comments.